nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-optimistic-locking/src/main/java/org/apache/nifi/web/revision/NaiveRevisionManager.java

```diff
@@ -423,7 +440,7 @@ public boolean requestWriteLock(final Revision proposedRevision) throws ExpiredR
                         throw ise;
                     }
 
-                    if (stamp.getClientId() == null || stamp.getClientId().equals(proposedRevision.getClientId())) {
+                    if (stamp.getUser() == null || stamp.getUser().equals(user)) {
```

Comment:
I think we need to be checking both the user and the client id here. To protect against the case when the same user has multiple clients open (like multiple browser windows). One client shouldn't be able to assume the transaction from another client. Should also have a unit test for this scenario.